oca/api.py

```diff
@@ -49,15 +32,66 @@ def method(self):
 
             [1, 2, 3]
 
+        If no container is provided as an argument to the decorator, and a
+        value is returned from the decorated method, a ValueError will be
+        raised.
+
+        .. code-block:: python
+
+            @api.foreach
+            def method(self):
+                # Raises ValueError
+                return 1+2
+
+            @api.foreach
+            def method(self):
+                # No error raised
+                self.math = 1+2
+
+    Args:
+        container (type): Type of container to wrap results in. It will be
+            instantiated with an iterable of the decorated method results.
+
+    Raises:
+        ValueError: If no container type is provided, and a value is returned
+            from the decorated method.
+        TypeError: If an incorrect data type is provided as the `container`.
+            Valid arguments are `None`, instances of `type`, or a callable.
+
     Returns:
         list: Of the return values from the decorated method, in the same
```

Comment:
This is wrong now. Returns whatever `container` returns when passing him a sorted generator.